src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -331,7 +333,8 @@ private static void collectDeclarations(Deque<AbstractFrame> frameStack, DetailA
                 collectVariableDeclarations(ast, frame);
                 break;
             case TokenTypes.PARAMETER_DEF :
-                if (!CheckUtils.isReceiverParameter(ast)) {
```

Comment:
@MEZk ! Actually, like I mentioned above, there was  a hidden problem in the logic, the lambda parameter in braces were getting collected as **local variables** in `beginTree` and hence since we fixed [3423](https://github.com/checkstyle/checkstyle/issues/3423) you wouldn't get any violations for them. Before that, you shall even get violations for lambda parameters in braces. So even though the behaviour is correct for lambda parameters in braces, the logic is wrong.